framework/aster-frame/src/task/scheduler.rs

```diff
@@ -60,3 +62,40 @@ pub fn fetch_task() -> Option<Arc<Task>> {
 pub fn add_task(task: Arc<Task>) {
     GLOBAL_SCHEDULER.lock_irq_disabled().enqueue(task);
 }
+
+/// A simple FIFO (First-In-First-Out) task scheduler.
+pub struct FifoScheduler {
+    /// A thread-safe queue to hold tasks waiting to be executed.
+    task_queue: SpinLock<VecDeque<Arc<Task>>>,
+}
+
+impl FifoScheduler {
+    /// Creates a new instance of `FifoScheduler`.
+    pub fn new() -> Self {
+        FifoScheduler {
+            task_queue: SpinLock::new(VecDeque::new()),
+        }
+    }
+}
+
+impl Default for FifoScheduler {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Scheduler for FifoScheduler {
+    /// Enqueues a task to the end of the queue.
+    fn enqueue(&self, task: Arc<Task>) {
+        self.task_queue.lock_irq_disabled().push_back(task.clone());
```

Comment:
`push_back(task)` is enough, `task.clone` is redudant.

Reply:
I notice that `switch_to_task` already uses a `clone` before `enqueue`, so `task.clone` is redundant. 
Moreover, `PreemptScheduler` may also need similar modifications.